core/controllers/topic_viewer_test.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2018 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for the topic viewer page."""
+
+from core.domain import story_domain
+from core.domain import story_services
+from core.domain import topic_domain
+from core.domain import topic_services
+from core.domain import user_services
+from core.tests import test_utils
+import feconf
+
+
+class BaseTopicViewerControllerTest(test_utils.GenericTestBase):
+
+    def setUp(self):
+        """Completes the sign-up process for the various users."""
+        super(BaseTopicViewerControllerTest, self).setUp()
+        self.signup(self.ADMIN_EMAIL, self.ADMIN_USERNAME)
+        self.admin_id = self.get_user_id_from_email(self.ADMIN_EMAIL)
+        self.set_admins([self.ADMIN_USERNAME])
+        self.admin = user_services.UserActionsInfo(self.admin_id)
+
+        self.topic_id = 'topic'
+        self.story_id = 'story'
+        self.topic_id_1 = 'topic1'
+
+        self.story = story_domain.Story.create_default_story(
```

Comment:
It's not an issue, but in the future, you could use the create_story (or similarly named, not sure) function in test_utils itself. So, you can do self.create_story() and pass in values as arguments.
Just a note for the future :).

Reply:
Will keep in mind thanks!